src/freenet/pluginmanager/PluginManager.java

```diff
@@ -1601,10 +1608,15 @@ public void addStartingPlugin(PluginProgress pluginProgress) {
 			}
 		}
 
+		/**
+		 * @return a copy of the starting plugins. Do not modify this: modifications will get thrown away.
+		 */
 		public Collection<PluginProgress> getStartingPlugins() {
+			Set<PluginProgress> startingPluginsCopy;
 			synchronized (this) {
-				return startingPlugins;
+				startingPluginsCopy = Set.copyOf(startingPlugins);
```

Comment:
Set.copyOf requires Java 10.
This is an error when using Java 8.

Reply:
Thank you! I fixed the setting of the IDE and pushed a change into next right-away (because this broke the build for Java 8 users whom we still support): https://github.com/hyphanet/fred/commit/4c3ffb183452314cc777d69a90418a73d095adf8